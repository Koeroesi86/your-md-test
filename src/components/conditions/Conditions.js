import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Condition from '../condition/Condition'
import LoadingIndicator from '../loading-indicator/LoadingIndicator'
import './Conditions.scss'

export function Conditions({ isLoaded, listing }) {
  return (
    <div className="Conditions">
      <h1>Conditions</h1>
      <div className="listing">
        {!isLoaded && (
          <LoadingIndicator />
        )}
        {!!isLoaded && listing.map((condition, index) => (
          <Condition key={`condition-${index}`} condition={condition}/>
        ))}
      </div>
    </div>
  )
}

Conditions.defaultProps = {
  isLoaded: false,
  listing: [],
};

Conditions.propTypes = {
  isLoaded: PropTypes.bool,
  listing: PropTypes.arrayOf(PropTypes.shape()),
};

function mapStateToProps(state) {
  return {
    isLoaded: state.conditions.loaded,
    listing: state.conditions.listing,
  }
}

function mapDispatchToProps(dispatch) {
  return {}
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Conditions)
