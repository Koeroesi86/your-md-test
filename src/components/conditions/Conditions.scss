.Conditions {
  .listing {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;

    .Condition {
      flex: 0 1 25%;

      @media (max-device-width: 660px) {
        flex-basis: 33%;
      }

      @media (max-device-width: 480px) {
        flex-basis: 50%;
      }

      @media (max-device-width: 320px) {
        flex-basis: 100%;
      }
    }

    .LoadingIndicator {
      flex: 1 1 0;
    }
  }
}
