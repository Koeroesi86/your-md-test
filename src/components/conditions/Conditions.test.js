import React from 'react'
import { shallow } from 'enzyme'
import { Conditions } from './Conditions'

describe('The Home component', () => {
  it('renders', () => {
    const result = shallow(<Conditions/>);

    expect(result).toMatchSnapshot();
  });
});
