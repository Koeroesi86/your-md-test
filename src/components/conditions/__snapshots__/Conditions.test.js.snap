// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home component renders 1`] = `
<div
  className="Conditions"
>
  <h1>
    Conditions
  </h1>
  <div
    className="listing"
  >
    <LoadingIndicator />
  </div>
</div>
`;
